### PR TITLE
Adds the brain damage spell for Wizards.

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
THE IDEA
[tested]
Adds the brain damage spell for wizards!
- A spell that deals 30 brain damage with a base cooldown of 20 seconds, lowered by 4 seconds for each cooldown upgrade down to 8 seconds;
- A power upgrade that allows it to deal 90 brain damage;
- 2 new icons, for normal and upgraded spell;
- Multiple invocations available.

MANY thanks to @MadmanMartian for helping a ton with the PR!

:cl:
 * rscadd: Adds the brain damage spell for wizards. Allows you to deal brain damage to whoever you target, with a base cooldown of 20 seconds that can be lowered to 8. Can be upgraded to triple the brain damage caused.

### CODE REVIEW

**Comment by Caboozles** [comment]
Is this physical brain damage or just regular?

**Comment by boy2mantwicethefam** [comment]
It's not organ damage, it's regular brain damage

File: code/modules/spells/targeted/retard.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	switch(spell_levels[Sp_POWER]) //If 0 it's normal, if 1 it's upgraded
+		if(0) //Not upgraded
+			name = "Brain Damage" //Nothing happens. Everything is already written above.
+		if(1) //Upgraded
+			name = "Brian Damage"
+			desc = "Give an unlucky person A LOT of brain damage."
+			amt_dam_brain = 90 //90 brain damage
+			message = "<span class='danger'>Yu fell dumb dumb!1!!<span>" //Message changes when upgraded
+			hud_state = "wiz_retard1" //Icon changes when upgraded
+		else // No way anything is going to be done if it isn't 0 or 1.
```
Comment (ShiftyRail) [inline code/modules/spells/targeted/retard.dm]: That's not really needed

**Review by MadmanMartian — APPROVED** [review]

**Comment by Arthurdentist** [comment]
I'm going to come out and say that I believe this is a bad spell and it will be bad for the game. I suspect this will be incredibly unfun to be on the end of. Having a lot of brain damage makes you unable to operate machines or doors, will randomly stun you, and makes you entirely reliant upon other people. Spells that do things like this are usually kind of bad. Disintergrate was removed initially because it  removed people from a round with no realistic way of countering it. Cluwnes were removed from the staff of change for the same reason; if you hit someone and turned them into a cluwne it becomes impossible for that person to fight back whatsoever. 

I suspect this spell would have similar effects; especially in combination with other spells that we already have. Combined with an invisibility cloak this spell will be quite powerful, but because the spell is ranged and not based on adjacency, the one item that will really synergize with it is the scrying orb. With the orb, you'll be able to take this spell, hide in maint and give people brain damage through the walls, which will be perhaps one of the worst wizard gimmicks in a while. This actually works with other ranged spells, but at the same time our ranged spells are usually weaker, and have limited time effects that wear off by themselves. I.E, Blind and Disorient go away after you wait long enough. Horsemask doesn't go away, but its quite a bit more trivial to fix compared with brain damage.

tl;dr, I really don't think this spell should be merged, but if it was changed so that it was an adjacency spell like clown curse and not a ranged spell, then it wouldn't be as much of an issue.

**Comment by Caboozles** [comment]
I don't think brain damage is any harder to fix than a horse mask. You can mass produce and hand out alky pills much faster than handing out pacid doses to the face. If you know the wizard is casting this, just mass produce alky and this will force the wizard to either gravitate towards medbay (and force a confrontation in some form), or just make some people retarded in the open.

This isn't organ damage, like he said (unless it is, in which case 100% agree with your concerns).

**Comment by Dacendeth** [comment]
Tinfoil hat should be made to protect you from this, same way mindswap is already prevented.

**Comment by Lunstra** [comment]
Oh god this is a ranged spell? Christ, it's not a bad spell aside from that though imo, but I'd say it definitely needs the range cutting. Imagine if you could clown curse from 7 tiles away cause that'd be about as obnoxious as this will be.

**Comment by bathosbathtime** [comment]
I like the idea of this spell but I agree that 7 tile range is overly powerful

**Comment by boy2mantwicethefam** [comment]
I don't believe it should be adjacent because it isn't as strong as something like clown curse, the brain damage is debilitating but it doesn't do something permanent like adding the clumsy gene to people. However, I do agree that the range should be nerfed. It is now 3 tiles, not any lower or else the wizard will have to expose themselves too much to retaliation. Tinfoil hat now blocks the spell too.

**Comment by MadmanMartian** [comment]
This would be surprisingly strong, as dexterity_check checks if brain loss is less than or equal to 60.

One cast, and suddenly you can't use guns, chem dispensers, etc.

**Comment by boy2mantwicethefam** [comment]
I'm still 70% sure it's RNG whether you can use machineries or guns or not

**Comment by Kammerjunk** [comment]
[sometimes it is](https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/machinery/doors/door.dm#L124)
[sometimes it isn't](https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/modules/power/apc.dm#L859)
Depends on the machinery, but most cases where it's RNG directly check `getBrainLoss()` whereas calls to `dexterity_check()` usually don't have any randomness

**Review by MadmanMartian — DISMISSED** [review]
Conflicts

**Comment by MadmanMartian** [comment]
adding the new icon won't stop the conflict, as it's not checking the contents of the files, but its last change.

Go to your git client, set your tracked branch to this repo's bleeding-edge, pull, rebase. There should be a conflict with this very file. Take their version, then once the rebase is complete, re-add your icons.

**Comment by boy2mantwicethefam** [comment]
I don't actually know what I did but everything looks fine now
